rotkehlchen/db/taxable_events.py

```diff
@@ -0,0 +1,348 @@
+            )
+            pass
+
+        return report
+
+    def get_events(self, filter_: ReportsFilterQuery) -> List[Dict[str, Any]]:
+        cursor = self.db.conn_transient.cursor()
+        query, bindings = filter_.prepare()
+        query = 'SELECT data FROM pnl_events ' + query
+        results = cursor.execute(query, bindings)
+
+        events = []
+        for result in results:
+            log.debug(f"get_events result: {result}")
+            try:
+                event = deserialize_event_from_db(result[0])
+            except DeserializationError as e:
+                self.msg_aggregator.add_error(
+                    f'Error deserializing PnL Event for Report from the DB. Skipping it.'
+                    f'Error was: {str(e)}',
+                )
+                continue
+
+            events.append(event)
+
+        return events
+
+    def add_report(self, start_ts: Timestamp, end_ts: Timestamp) -> int:
+        cursor = self.db.conn_transient.cursor()
+        timestamp = ts_now()
+        query = """
+        INSERT INTO pnl_reports(
+            name, timestamp, start_ts, end_ts
+        )
+        VALUES (?, ?, ?, ?)"""
+        cursor.execute(query, (f"Report from {start_ts} to {end_ts}", timestamp, start_ts, end_ts))
+        identifier = cursor.lastrowid
+        self.db.conn_transient.commit()
+        return identifier
+
+    def add_event(self, report_id: int, time: Timestamp, event: dict) -> None:
+        """Adds a new event to a transient report for the PnL history in a given time range
+
+        May raise:
+        - sqlcipher.IntegrityError if there is a conflict at serialization of the event
+        """
+        cursor = self.db.conn_transient.cursor()
+        query = """
+        INSERT INTO pnl_events(
+            report_id, timestamp, data
+        )
+        VALUES(?, ?, ?);"""
+        cursor.execute(query, (report_id, time, serialize_event_to_db(event)))
+        self.db.conn_transient.commit()
+
+    def add_events(self, report_id: int, events: List[Dict[str, Any]]) -> None:
+        """Adds taxable events to the database"""
+        event_tuples: List[Tuple[Any, ...]] = []
+        for event in events:
+            event_tuples.append((
+                report_id,
+                event['time'],
+                serialize_event_to_db(event),
+            ))
+
+        query = """
+            INSERT INTO pnl_events(
+              report_id,
+              timestamp,
+              event
+            )
+            VALUES (?, ?, ?)
+        """
+        self.db.write_tuples(
+            tuple_type='pnl_event',
+            query=query,
+            tuples=event_tuples,
+        )
+
+    def get_or_query_report_events(
+            self,
+            report_id: int,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> Tuple[List[Dict[str, Any]], Optional[Timestamp], Optional[Timestamp]]:
+        """Only queries cached events and prepares them for the response"""
```

Comment:
I don't understand what this function does. So it queries the range of locations to query? Queries only cached data? Why even mess with location query ranges here? I don't understand.

Why retrun the last timestamps you got from location_query_ranges? It makes no sense.